### PR TITLE
Add Mxnet MNIST example

### PR DESCRIPTION
**Pull request by hsl89**
*Issue #, if available:*

*Description of changes:*
1. Add notebooks for training and deployment on MNIST dataset with MXnet gluon
2. Download data set directly from S3 bucket

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

File: frameworks/mxnet/code/inference.py [hunk 1]
```diff
@@ -0,0 +1,61 @@
+from __future__ import print_function
```
Comment (aaronmarkham) [inline frameworks/mxnet/code/inference.py:1]: Please add the apache 2.0 license header here

File: frameworks/mxnet/code/test_inference.py [hunk 2]
```diff
@@ -0,0 +1,90 @@
+from inference import transform_fn, model_fn
```
Comment (aaronmarkham) [inline frameworks/mxnet/code/test_inference.py:1]: needs license

File: frameworks/mxnet/code/test_train.py [hunk 3]
```diff
@@ -0,0 +1,60 @@
+from train import train, parse_args
```
Comment (aaronmarkham) [inline frameworks/mxnet/code/test_train.py:1]: needs license

File: frameworks/mxnet/code/train.py [hunk 4]
```diff
@@ -0,0 +1,198 @@
+from __future__ import print_function
```
Comment (aaronmarkham) [inline frameworks/mxnet/code/train.py:1]: needs license

**Review by aaronmarkham — CHANGES_REQUESTED** [review]

File: frameworks/mxnet/code/inference.py (hunk 1 above)
Comment (hsl89) [reply frameworks/mxnet/code/inference.py:18]: License added in all .py files in .`/code`

File: frameworks/mxnet/code/test_inference.py [hunk 5]
```diff
@@ -0,0 +1,105 @@
...
+            s3.Bucket(bucket_name).download_file(
+                key, os.path.join(model_dir, 'model.tar.gz'))
+        except botocore.exceptions.ClientError as e:
+            if e.response['Error']['Code'] == '404':
+                print("the object does not exist.")
+            else:
+                raise
+
+    # untar the model
+    tar = tarfile.open(os.path.join(model_dir, 'model.tar.gz'))
```
[the remark this answers is not in the document]
Comment (hsl89) [reply frameworks/mxnet/code/test_inference.py:64]: nice

**Pull request merged**